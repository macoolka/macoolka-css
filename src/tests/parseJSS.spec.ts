
import {  parseStand as parse} from '..';

describe('parse', () => {
    it('parse stand', () => {
        expect(parse({
            mkPadding: 'small',
            height: 100,
            selector: {
                '&:focus': {
                    mkPadding: 'medium',
                    height: 200,
                }
            }
        })).toMatchSnapshot();
    });
    it('parse children', () => {
        expect(parse({
            mkPadding: 'small',
            mkChildren: {
                mkHoverBorderColor:'main',
                minWidth:20,
                selector:{
                    ':focus':{
                        color:'red'
                    }
                }
            },
            selector: {
                '&:focus': {
                    mkPadding: 'medium',
                    height: 200,
                }
            }
        })).toMatchSnapshot();
    });
    it('parse media children', () => {
        expect(parse({
            mkPadding: 'small',
            mkMedia: [{
                mkChildren: {
                    mkHoverBorderColor:'main',
                    minWidth:20,
                    selector:{
                        ':focus':{
                            color:'red'
                        }
                    }
                },
            }, {
                mkChildren: {
                    mkHoverBorderColor:'accent',
                    minWidth:10,
                    selector:{
                        ':focus':{
                            color:'green'
                        }
                    }
                },
            }] ,

            selector: {
                '&:focus': {
                    mkPadding: 'medium',
                    height: 200,
                }
            }
        })).toMatchSnapshot();
    });
    it('parse media', () => {
        expect(parse({
            height: 100,
            mkMedia: [{
                mkPadding: 'small',
                fontSize: 12
            }, {
                mkPadding: 'medium', fontSize: 13
            }, {
                mkPadding: 'large', fontSize: 14
            }, {
                mkPadding: 'medium', fontSize: 15
            }] ,
            selector: {
                '&:focus': {
                    mkColor: 'main',
                    borderWidth: 1
            }
            }
        })).toMatchSnapshot();
    });
    it('parse others prop', () => {
        expect(parse({
            height: 100,
            onclick:15,
            mkMedia: [{
                mkPadding: 'small',
                fontSize: 12
            }, {
                mkPadding: 'medium', fontSize: 13
            }, {
                mkPadding: 'large', fontSize: 14
            }, {
                mkPadding: 'medium', fontSize: 15
            }] ,
            selector: {
                '&:focus': {
                    mkColor: 'main',
                    borderWidth: 1
            }
            }
        } as any )).toMatchSnapshot();
    });

});