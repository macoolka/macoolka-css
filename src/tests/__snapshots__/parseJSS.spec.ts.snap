// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse parse children 1`] = `
"padding: 8px;
&:focus {
  padding: 16px;
  height: 200px;
}
> * {
  min-width: 20px;
}
> *:focus {
  color: red;
}
> *:hover:not(:disabled) {
  border-color: #00796b;
}
"
`;

exports[`parse parse media 1`] = `
"height: 100px;
&:focus {
  border-width: 1px;
  background-color: #26a69a;
  color: #fff;
}
@media screen and (max-width: 120em) {
  font-size: 15px;
  padding: 16px;
}
@media screen and (max-width: 93em) {
  font-size: 14px;
  padding: 24px;
}
@media screen and (max-width: 80em) {
  font-size: 13px;
  padding: 16px;
}
@media screen and (max-width: 50em) {
  font-size: 12px;
  padding: 8px;
}
"
`;

exports[`parse parse media children 1`] = `
"padding: 8px;
&:focus {
  padding: 16px;
  height: 200px;
}
@media screen and (max-width: 80em) {
}
@media screen and (max-width: 50em) {
}
@media screen and (max-width: 80em) {
  > * {
    min-width: 10px;
  }
}
@media screen and (max-width: 80em) {
  > *:hover:not(:disabled) {
    border-color: #f50057;
  }
}
@media screen and (max-width: 80em) {
  > *:focus {
    color: green;
  }
}
@media screen and (max-width: 50em) {
  > * {
    min-width: 20px;
  }
}
@media screen and (max-width: 50em) {
  > *:hover:not(:disabled) {
    border-color: #00796b;
  }
}
@media screen and (max-width: 50em) {
  > *:focus {
    color: red;
  }
}
"
`;

exports[`parse parse others prop 1`] = `
"height: 100px;
onclick: 15;
&:focus {
  border-width: 1px;
  background-color: #26a69a;
  color: #fff;
}
@media screen and (max-width: 120em) {
  font-size: 15px;
  padding: 16px;
}
@media screen and (max-width: 93em) {
  font-size: 14px;
  padding: 24px;
}
@media screen and (max-width: 80em) {
  font-size: 13px;
  padding: 16px;
}
@media screen and (max-width: 50em) {
  font-size: 12px;
  padding: 8px;
}
"
`;

exports[`parse parse stand 1`] = `
"padding: 8px;
height: 100px;
&:focus {
  padding: 16px;
  height: 200px;
}
"
`;
